File: packages/contract-py/src/tigr_asgi_contract/framing.py
from ._enum_compat import StrEnum


class Framing(StrEnum):
    JSON = 'json'
    JSONRPC = 'jsonrpc'
    NDJSON = 'ndjson'
    SSE = 'sse'
    TEXT = 'text'
    BYTES = 'bytes'
    BINARY = 'binary'

File: packages/contract-rs/src/framing.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub enum Framing {
    #[serde(rename = "json")]
    Json,
    #[serde(rename = "jsonrpc")]
    Jsonrpc,
    #[serde(rename = "ndjson")]
    Ndjson,
    #[serde(rename = "sse")]
    Sse,
    #[serde(rename = "text")]
    Text,
    #[serde(rename = "bytes")]
    Bytes,
    #[serde(rename = "binary")]
    Binary,
}
